tests/benchmark/crud_test.go

```diff
@@ -0,0 +1,920 @@
+				},
+			}
+			// Only take reading for the first version
+			if j == 1 {
+				b.StartTimer()
+			}
+			_, err := client.UpdateApiVersion(ctx, req)
+
+			if j == 1 {
+				b.StopTimer()
+			}
+			if i > 1 && err != nil {
+				b.Errorf("UpdateApiVersion(%+v) returned unexpected error: %s", req, err)
+			}
+		}
+	}
+}
+
+func BenchmarkUpdateApiSpecVersion(b *testing.B) {
+	rootResource := getRootResource(b)
+
+	ctx := context.Background()
+	gapicClient, err := connection.NewClient(ctx)
+	if err != nil {
+		b.Fatalf("Setup: Failed to create client: %s", err)
+	}
+	client := gapicClient.GrpcClient()
+
+	b.StopTimer()
+	b.ResetTimer()
+
+	for i := 1; i <= b.N; i++ {
+		pauseAfter1000Iterations(i)
+
+		apiId := getApiName(i, b)
+
+		for j := 1; j <= loadVersionCount(b); j++ {
+			req := &rpc.UpdateApiSpecRequest{
+				ApiSpec: &rpc.ApiSpec{
+					Name:        fmt.Sprintf("%s/apis/%s/versions/v%d/specs/spec-%d", rootResource, apiId, j, j),
+					Description: fmt.Sprintf("Updated description of spec-%d for v%d of %s", j, j, apiId),
+				},
+				UpdateMask: &field_mask.FieldMask{
+					Paths: []string{"description"},
+				},
+			}
+			// Only take reading for the first version
+			if j == 1 {
+				b.StartTimer()
+			}
+			_, err := client.UpdateApiSpec(ctx, req)
+			if j == 1 {
+				b.StopTimer()
+			}
+			if i > 1 && err != nil {
+				b.Errorf("UpdateApiSpec(%+v) returned unexpected error: %s", req, err)
+			}
+		}
+	}
+}
+
+func BenchmarkUpdateArtifact(b *testing.B) {
+	rootResource := getRootResource(b)
+
+	ctx := context.Background()
+	gapicClient, err := connection.NewClient(ctx)
+	if err != nil {
+		b.Fatalf("Setup: Failed to create client: %s", err)
+	}
+	client := gapicClient.GrpcClient()
+
+	b.StopTimer()
+	b.ResetTimer()
+
+	for i := 1; i <= b.N; i++ {
+		pauseAfter1000Iterations(i)
+
+		apiId := getApiName(i, b)
+
+		for j := 1; j <= loadVersionCount(b); j++ {
+			messageMimeType := "text/plain"
+			artifactId := "self-parent-link"
+			messageContents := []byte(fmt.Sprintf("Updated - %s", artifactId))
+			req := &rpc.ReplaceArtifactRequest{
+				Artifact: &rpc.Artifact{
+					Name:     fmt.Sprintf("%s/apis/%s/versions/v%d/specs/spec-%d/artifacts/%s", rootResource, apiId, j, j, artifactId),
+					MimeType: messageMimeType,
+					Contents: messageContents,
+				},
+			}
+			// Only take reading for the first version
+			if j == 1 {
+				b.StartTimer()
+			}
+			_, err = client.ReplaceArtifact(ctx, req)
+			if j == 1 {
+				b.StopTimer()
+			}
+			if i > 1 && err != nil {
+				b.Errorf("ReplaceArtifact(%+v) returned unexpected error: %s", req, err)
+			}
+		}
+	}
+}
+
+func BenchmarkGetApi(b *testing.B) {
+	rootResource := getRootResource(b)
+
+	ctx := context.Background()
+	gapicClient, err := connection.NewClient(ctx)
+	if err != nil {
+		b.Fatalf("Setup: Failed to create client: %s", err)
+	}
+	client := gapicClient.GrpcClient()
+	b.StopTimer()
+	b.ResetTimer()
+
+	for i := 1; i <= b.N; i++ {
+		pauseAfter1000Iterations(i)
```

Comment:
Why are we pausing between GET calls? The comment in the pause function says we only need throttling for mutations.

Reply:
Removed the throttling from the GET / LIST calls